### PR TITLE
[1/3] DXCDT-345: Update golanci lint config and fix issues

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

A little maintenance task to get the project in better shape by updating the golang ci lint configuration and fixing any raised issues. 

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -1,21 +1,47 @@
...
+    rules:
+      - name: package-comments
+        severity: warning
+        disabled: true
 
 issues:
   exclude-use-default: false
+  exclude:
+    - "Error return value of `response.Body.Close` is not checked"
+    - "Error return value of `w.Write` is not checked"
```
Comment (sergiught) [inline .golangci.yaml:42]: We can't do much with the error a potential `defer response.Body.Close()` raises. The func returns an error mostly to ensure it follows the `Closer` interface. Same goes for `w.Write(), so we're just gonna not enforce linting over them.

File: doc.go [hunk 2]
```diff
@@ -3,75 +3,74 @@ Package auth0 provides a client for using the Auth0 Management API.
 
 Usage
 
-    import (
-        github.com/auth0/go-auth0
-        github.com/auth0/go-auth0/management
-    )
+	import (
```
Comment (sergiught) [inline doc.go:6]: We'll update the content of this page in a follow up PR.

File: management/example_test.go [hunk 3]
```diff
@@ -1,181 +0,0 @@
-package management_test
```
Comment (sergiught) [inline management/example_test.go:1]: Removing the examples as they don't provide any usefulness inside the godoc pages as they're not runnable without the proper credentials so they won't display any output: https://pkg.go.dev/github.com/auth0/go-auth0@v0.14.0/management#pkg-examples. 

For more info:
- https://go.dev/blog/examples

File: management/prompt.go [hunk 4]
```diff
@@ -19,10 +19,6 @@ type PromptManager struct {
 	*Management
 }
 
-type promptCustomText struct {
```
Comment (sergiught) [inline management/prompt.go:22]: This was never used anywhere and it's also a private type, so it can safely be removed.

**Review by willvedd — APPROVED** [review]
Looks good 👍

**Pull request merged**